### PR TITLE
testing/py3-pytest-timeout: new aport

### PR DESCRIPTION
**Pull request by OTLabs**
https://pypi.org/project/pytest-timeout/
py.test plugin to abort hanging tests

### CODE REVIEW

**Comment by OTLabs** [comment]
Same tests fail on all architectures:

```
=========================== short test summary info ============================
--
457 | FAILED test_pytest_timeout.py::test_fix_setup[function-meth0]
458 | FAILED test_pytest_timeout.py::test_fix_setup[class-meth0]
459 | FAILED test_pytest_timeout.py::test_fix_setup[module-meth0]
460 | FAILED test_pytest_timeout.py::test_fix_setup[session-meth0]
461 | FAILED test_pytest_timeout.py::test_fix_finalizer[function-meth0]
462 | FAILED test_pytest_timeout.py::test_fix_finalizer[class-meth0]
463 | FAILED test_pytest_timeout.py::test_fix_finalizer[module-meth0]
464 | FAILED test_pytest_timeout.py::test_fix_finalizer[session-meth0]
465 | ===================== 8 failed, 27 passed in 42.96 seconds =====================
```

**Comment by OTLabs** [comment]
Reported upstream https://bitbucket.org/pytest-dev/pytest-timeout/issues/37/

**Comment by TBK** [comment]
Shouldn't this be named `py3-pytest-timeout`?

**Comment by OTLabs** [comment]
@TBK Fixed. Thank you!

I have disables 8 failing tests.

**Comment by OTLabs** [comment]
@maxice8 I have removed the LICENSE.

I have also disabled the `python install...` in `check()` and tested with PYTHONPATH="$PWD/build:$PYTHONPATH" and PYTHONPATH="$PWD/build/lib:$PYTHONPATH". Both of them produce massive test fails.

**Comment by OTLabs** [comment]
Thanks to @5paceToast for `tox` solution!

**Comment by 5paceToast** [comment]
I'd also recommend testing if using `tox -s true \` in place of `tox -e python3.7 \` will work.
As-is this package will start failing once python 3.8 comes out and this package is rebuilt, unless one pays attention.

**Comment by OTLabs** [comment]
It doesn't.

**Comment by Ikke** [comment]
Somehow this is failing to build for me locally. Tox gives this error:

> ERROR: unknown environment 'python3.7'

If I look at tox.ini, I see this:

```ini
envlist = py27,py35,py36,pypy,pypy3
```

then I don't see python3.7 mentioned there. If I add it, the tests seem to go through.

File: testing/py3-pytest-timeout/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+source="https://files.pythonhosted.org/packages/13/48/7a166eaa29c1dca6cc253e3ba5773ff2e4aa4f567c1ea3905808e95ac5c1/pytest-timeout-$pkgver.tar.gz"
+builddir="$srcdir/pytest-timeout-$pkgver"
+
+build() {
+	python3 setup.py build
+}
+
+check() {
+	tox -e python3.7 -- \
+		--deselect=test_pytest_timeout.py::test_fix_setup[function-meth0] \
```
Comment (Ikke) [inline testing/py3-pytest-timeout/APKBUILD]: shellcheck warns that the `[]` in here are interpreted as character classes. Perhaps escape/quote them?

Comment (OTLabs) [reply testing/py3-pytest-timeout/APKBUILD]: Done! Thank you!

**Comment by OTLabs** [comment]
I have switched from `tox -e python3.7` to `tox -e py37` and now it runs fine without any modifications to `tox.ini`.

The error might be related with recent upgrade to `tox`.

**Comment by graingert** [comment]
@OTLabs @maxice8 the upstream issue is now at https://github.com/pytest-dev/pytest-timeout/issues/37

does it still apply?

**Comment by OTLabs** [comment]
@graingert I have tested with `pytest-timeout 1.3.4`.

I get no any errors running `tox -e py38` , but if I run `tox --sitepackages -e py38` I get 32 failed tests and 3 passed.

**Pull request closed without merging**